### PR TITLE
Feature/cart query

### PR DESCRIPTION
**Pull request by dendoesit**
## Describe your changes

Added cart query

## Justify why they are needed

To get product data automatically.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

[GRW-1328]: https://hedvig.atlassian.net/browse/GRW-1328?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by simonauner — APPROVED** [review]
Nice! The pieces are falling into place 🤩

File: apps/store/src/components/CartPage/CartPage.tsx [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
 import { PageLink } from '@/lib/PageLink'
+import { CartPageProps } from './CartPageProps.types'
 
-export const CartPage = () => {
-  const products = [
-    { name: 'Home Insurance', cost: 250, currency: 'SEK' },
-    { name: 'Apartment Insurance', cost: 100, currency: 'SEK' },
-  ]
-  const cost = { total: 350, subTotal: 250 }
+export const CartPage = (props: CartPageProps) => {
```
Comment (robinandeer) [inline apps/store/src/components/CartPage/CartPage.tsx:9]: nit: we can destructure directly here :)

File: apps/store/src/components/PriceBreakdown.tsx/PriceBreakdown.tsx [hunk 2]
```diff
@@ -22,21 +21,23 @@ export const PriceBreakdown = ({ currency, cost, products, campaigns }: Props) =
           {products.map((product) => (
             <DataRow key={product.name}>
               <Text size="m">{product.name}</Text>
-              <Price>{currencyFormatter.format(product.cost)}</Price>
+              <Price>{currencyFormatter.format(parseInt(product.cost))}</Price>
```
Comment (robinandeer) [inline apps/store/src/components/PriceBreakdown.tsx/PriceBreakdown.tsx:24]: nit: would probably be better to pass cost as a number if that's what the React component expects

also maybe safer to "parseFloat" so we don't make rounding errors (even if we currently don't charge in decimals -- but maybe when we have EUR)

**Review by robinandeer — APPROVED** [review]
@simonauner totally agree, we are adding this to the cart in the API!

**Review by robinandeer — COMMENTED** [review]
Error from Vercel build:

```
store:codegen: ./src/pages/cart/index.tsx
store:codegen: 6:10  Error: 'getShopSessionServerSide' is defined but never used.  @typescript-eslint/no-unused-vars
```

File: apps/store/src/components/PriceBreakdown.tsx/PriceBreakdown.tsx (hunk 2 above)
Comment (dendoesit) [reply apps/store/src/components/PriceBreakdown.tsx/PriceBreakdown.tsx:24]: this is coming from the ShopSessionQuery as a string.   -->       price: { __typename?: 'Money'; amount: string; currencyCode: CurrencyCode }

File: apps/store/src/components/PriceBreakdown.tsx/PriceBreakdown.tsx [hunk 3]
```diff
@@ -22,21 +21,23 @@ export const PriceBreakdown = ({ currency, cost, products, campaigns }: Props) =
           {products.map((product) => (
             <DataRow key={product.name}>
               <Text size="m">{product.name}</Text>
-              <Price>{currencyFormatter.format(product.cost)}</Price>
+              <Price>{currencyFormatter.format((product.cost))}</Price>
```
Comment (robinandeer) [inline apps/store/src/components/PriceBreakdown.tsx/PriceBreakdown.tsx:24]: nit: looks like an extra parentheses here

**Review by robinandeer — APPROVED** [review]

**Pull request merged**